system/jlib/jlog.cpp

```diff
@@ -327,6 +333,32 @@ const LogMsgJobInfo & checkDefaultJobInfo(const LogMsgJobInfo & _jobInfo)
     return _jobInfo;
 }
 
+const LogMsgTraceInfo & queryActiveTraceInfo()
+{
+    if (defaultTraceInfoId != UnknownTraceInfoId)
+    {
+        const LogMsgTraceInfo * traceInfo = theManager->queryTraceInfo(defaultTraceInfoId);
+        if (traceInfo)
+            return *traceInfo;
+    }
+
+    return unknownTrace;
+}
+
+LogMsgTraceInfoId setDefaultTraceInfo(const char *theTraceID, const char *theSpanID, bool threaded)
+{
+    return setDefaultTraceInfo(theManager->addTraceInfo(theTraceID, theSpanID), threaded);
```

Comment:
The code would be cleaner/more symmetric if this addTraceInfo() call was in the CSpan (to match the remove call).  However if everything turns out cleaner storing an ISpan in the thread context then that will remove the need for this add.